TA_Scripts/Update_Students/createNewYml.js

```diff
@@ -0,0 +1,69 @@
+/*
+  Developed by Ten Zhi-Yang, 2017
+  To run this script, run:
+  node createNewYml.js
+
+  it will generate a new students.yml in the same folder for you to manually update. (instead of replacing the original file)
+  in order to follow existing standards, it will also generate the image id in lowercase and hyphens. The images will have to be renamed manually
+
+  Disclaimer: file follows the Javascript Standard Style, so if you don't like the lack of semi-colons, too bad!
+*/
+
+// generate csv from google docs and put in the same folder, update this name
+var csvFilename = 'contactlist.csv'
+// depending on the csv generated you may need to change the key (title in the csv)
+var nameKey = 'Name'
+var facKey = 'Faculty'
+var blogKey = 'Blog'
+var oneLinerKey = 'One-liner'
+
+// Unless yml format or dir change, there should be no need to modify this
+var originalFileDir = '../../_data/students.yml'
+var csvToJson = require('./dependancies/csv.js')
+var fs = require('fs')
+
+var obj = {
+  filename: csvFilename
+}
+
+var callback = function (err, json) {
+  if (err) {
+    return console.error(err)
+  }
+  // console.log(json)
+  json.map(function (value) {
+    value.id = value[nameKey].toLowerCase().replaceAll(' ', '-')
```

Comment:
This operation is called slugify. Doing lower case and replacing spaces with hyphen won't cover all possible cases that might be present in future batches. Would recommend to use a node package to do the slugification instead.